File: patterns/two-pointers/02.remove-duplicates-from-sorted-array-2/main.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i, limit = 0, 2

        for num in nums:
            if i < limit or nums[i-limit] != num:
                nums[i] = num
                i += 1
        
        return i

File: patterns/two-pointers/02.remove-duplicates-from-sorted-array-2/main.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0, limit = 2;

        for (auto num : nums) {
            if ( i < limit || nums[i-limit] != num) {
                nums[i] = num;
                i++;
            }
        }

        return i;
    }
};
